nd4j/nd4j-backends/nd4j-tests/src/test/java/org/nd4j/linalg/factory/ops/NDBaseTest.java

```diff
@@ -0,0 +1,923 @@
+/* *****************************************************************************
+ * Copyright (c) 2015-2018 Konduit k.k.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Apache License, Version 2.0 which is available at
+ * https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ ******************************************************************************/
+
+package org.nd4j.linalg.factory.ops;
+
+import org.junit.Test;
+import org.nd4j.linalg.BaseNd4jTest;
+import org.nd4j.linalg.api.buffer.DataType;
+import org.nd4j.linalg.api.ndarray.INDArray;
+import org.nd4j.linalg.factory.Nd4j;
+import org.nd4j.linalg.factory.Nd4jBackend;
+import org.nd4j.linalg.indexing.conditions.Conditions;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+public class NDBaseTest extends BaseNd4jTest {
+    public NDBaseTest(Nd4jBackend backend) {
+        super(backend);
+    }
+
+    @Test
+    public void testAll() {
+        NDBase base = new NDBase();
+
+        INDArray x = Nd4j.zeros(DataType.DOUBLE, 3, 3);
+        INDArray y = base.all(x, 1);
+        INDArray y_exp = Nd4j.create(new boolean[]{false, false, false});
+        assertTrue(y.equalsWithEps(y_exp, 1e-6));
+    }
+
+    @Test
+    public void testAny() {
+        NDBase base = new NDBase();
+
+        INDArray x = Nd4j.eye(3);
+        INDArray y = base.any(x, 1);
+        INDArray y_exp = Nd4j.create(new boolean[]{true, true, true});
+        assertTrue(y.equalsWithEps(y_exp, 1e-6));
+    }
+
+    @Test
+    public void testArgmax() {
+        NDBase base = new NDBase();
+
+        INDArray x = Nd4j.create(new double[][]{{0.75, 0.5, 0.25}, {0.5, 0.75, 0.25}, {0.5, 0.25, 0.75}});
+        INDArray y = base.argmax(x, 0); //with default keepdims
+        INDArray y_exp = Nd4j.createFromArray(0L, 1L, 2L);
+        assertEquals(y.dataType(), DataType.LONG);
```

Comment:
See comment above. But also expected always goes first, otherwise junit message is confusing:
```
java.lang.AssertionError: 
Expected : <wrong value>
Actual   : <what we actually want>
```

There's a whole lot of these.
Might seem like a nitpick, but I have definitely wasted time when debugging tests without realizing that the order is flipped in assertEquals...